app/helpers/schema_helper.py

```diff
@@ -0,0 +1,46 @@
+class SchemaHelper(object):
+
+    @staticmethod
+    def get_first_group_id(survey_json):
+        return survey_json['groups'][0]['id']
+
+    @staticmethod
+    def get_last_group_id(survey_json):
+        return survey_json['groups'][-1]['id']
+
+    @staticmethod
+    def get_first_block_id(survey_json):
+        return survey_json['groups'][0]['blocks'][0]['id']
+
+    @staticmethod
+    def get_blocks(survey_json):
+        blocks = []
+        for group in survey_json['groups']:
+            for block in group['blocks']:
+                blocks.append(block)
+        return blocks
+
+    @staticmethod
+    def get_groups(survey_json):
```

Comment:
Annoyingly, I originally implemented it this way and backtracked as I was having trouble getting it working with Navigator. Great minds eh? I've managed to get it working now though.